include/boost/fusion/container/vector/vector.hpp

```diff
@@ -9,12 +9,348 @@
+            BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            explicit
+            vector_data(each_elem, U&&... var)
+                : store<I, T>(std::forward<U>(var))...
+            {}
+
+            template <typename Sequence>
+            BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            void
+            assign(Sequence&&, detail::index_sequence<>) {}
+
+            template <typename Sequence, std::size_t N, std::size_t ...M>
+            BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            void
+            assign(Sequence&& seq, detail::index_sequence<N, M...>)
+            {
+                at_impl(mpl::int_<N>()) = vector_detail::forward_at_c<N>(seq);
+                assign(std::forward<Sequence>(seq), detail::index_sequence<M...>());
+            }
+
+            typedef extension::value_at_impl<vector_tag> value_at_impl;
+
+            template <typename J>
+            BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            typename value_at_impl::template apply<vector_data, J>::type&
+            at_impl(J)
+            {
+                typedef typename value_at_impl::template apply<vector_data, J>::type U;
+                typedef store<J::value, U> S;
+                return static_cast<S*>(this)->get();
+            }
+
+            template <typename J>
+            BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+            typename value_at_impl::template apply<vector_data, J>::type const&
+            at_impl(J) const
+            {
+                typedef typename value_at_impl::template apply<vector_data, J>::type U;
+                typedef store<J::value, U> S;
+                return static_cast<S const*>(this)->get();
+            }
+        };
+
+        // Internal use only specialization for implementing vectorN.
+        // Simple aliasing (like a using vectorN = vector<T...>) will cause specialization confliction
+        // like following.
+        //
+        //  template <typename> struct SomeClass;
+        //  template <typename T> struct SomeClass<vector<T>> { ... };
+        //  template <typename T>
+        //  struct SomeClass<vector1<T>> // Error, since vector<T> and vector1<T> are exact same type.
+        //  { ... };
+        //
+        // Introducing `numbered_vector_tag` will resolve such specialization error.
+        //
+        //  template <typename T>
+        //  struct SomeClass<vector<T>> { ... };
+        //  template <typename T>
+        //  struct SomeClass<vector1<T>> // OK
+        //  { ... };
+        //
+        //  // Same meaning as above specialization.
+        //  template <typename T>
+        //  struct SomeClass<vector<numbered_vector_tag<1>, T>>
+        //  { ... };
+        template <std::size_t ...I, std::size_t N, typename ...T>
+        struct vector_data<detail::index_sequence<I...>, numbered_vector_tag<N>, T...>
+            : vector_data<typename detail::make_index_sequence<sizeof...(T)>::type, T...>
```

Comment:
Looks like you could get rid of this entire specialization if you specialize `construct_vector` below.